pkg/resources/stream_on_directory_table_acceptance_test.go

```diff
@@ -0,0 +1,342 @@
+					HasStageString(stage.ID().Name()),
+					resourceshowoutputassert.StreamShowOutput(t, resourceName).
+						HasCreatedOnNotEmpty().
+						HasName(id.Name()).
+						HasDatabaseName(id.DatabaseName()).
+						HasSchemaName(id.SchemaName()).
+						HasOwner(snowflakeroles.Accountadmin.Name()).
+						HasTableName(stage.ID().Name()).
+						HasSourceType(sdk.StreamSourceTypeStage).
+						HasBaseTablesPartiallyQualified(stage.ID().Name()).
+						HasType("DELTA").
+						HasStale("false").
+						HasMode(sdk.StreamModeDefault).
+						HasStaleAfterNotEmpty().
+						HasInvalidReason("N/A").
+						HasComment("bar").
+						HasOwnerRoleType("ROLE"),
+					assert.Check(resource.TestCheckResourceAttrSet(resourceName, "describe_output.0.created_on")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.name", id.Name())),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.database_name", id.DatabaseName())),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.schema_name", id.SchemaName())),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.owner", snowflakeroles.Accountadmin.Name())),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.comment", "bar")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.table_name", stage.ID().Name())),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.source_type", string(sdk.StreamSourceTypeStage))),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.base_tables.#", "1")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.base_tables.0", stage.ID().Name())),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.type", "DELTA")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.stale", "false")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.mode", string(sdk.StreamModeDefault))),
+					assert.Check(resource.TestCheckResourceAttrSet(resourceName, "describe_output.0.stale_after")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.owner_role_type", "ROLE")),
+				),
+			},
+			// import
+			{
+				Config:       config.FromModel(t, modelWithExtraFieldsModified),
+				ResourceName: resourceName,
+				ImportState:  true,
+				ImportStateCheck: assert.AssertThatImport(t,
+					resourceassert.ImportedStreamOnDirectoryTableResource(t, resourceId).
+						HasNameString(id.Name()).
+						HasDatabaseString(id.DatabaseName()).
+						HasSchemaString(id.SchemaName()).
+						HasFullyQualifiedNameString(id.FullyQualifiedName()).
+						HasStageString(stage.ID().Name()).
+						HasCommentString("bar"),
+				),
+			},
+		},
+	})
+}
+
+func TestAcc_StreamOnDirectoryTable_CopyGrants(t *testing.T) {
```

Comment:
I'd say that this test only checks the proper suppression (it could be simplified with emptyplan checks, though), but I would rather see a longer test, that takes copy grants into consideration and we check it does (e.g. that with copy grants set to true and replacement, we still have the grants, and with set to false we don't).

Reply:
Okay, I'll keep it open then

Reply:
Added here.